src/modules/createInstantSearchRouting.js

```diff
@@ -22,7 +22,14 @@ export default ({ context, indexName }) => ({
             });
 
             if (typeof history === 'object') {
-                history.pushState(routeState, null, query || location.pathname);
+                // write query string if route would change only:
+                if (query && location.search !== query) {
+                    // query is different
+                    history.pushState(routeState, null, query);
+                } else (location.search) {
```

Comment:
@Haroenv

Is this supposed to be an `else if`?

Reply:
yes, sorry, that's supposed to be `else if`
```suggestion
                } else if (location.search) {
```